### DISCUSSION TITLE
individual a z t parameters are negative value when using hierarchical model

### DISCUSSION THREAD

**Discussion by YuanboBQ** [lnccbrown/HSSM#418] [General]
**HSSM version**
[v0.2.0b1](https://github.com/lnccbrown/HSSM/releases/tag/v0.2.0b1)

Hi,
Thanks for updating the package. I am currently working with the v0.2.0b1 version of HSSM package, and I have encountered an issue while handling my own data. Specifically, when using hierarchical models with the sampler set to "nuts_numpyro", I am facing the following problem: 
When I did not set `link_settings = "log_logit"`, the sampling process was slow, and resulting in high values for r_hat.  When I  set `link_settings = "log_logit"`, the sampling process was normal, and the r_hat was below 1.05. But the individual a z t parameters are negative value (See the below fig).  How to fix this issue? Thanks!
<img width="852" alt="截屏2024-01-16 09 22 14" src="https://github.com/lnccbrown/HSSM/assets/2658686/96bda3a6-8358-48f6-8391-fc46c4bacc85">
<img width="851" alt="截屏2024-01-16 09 22 33" src="https://github.com/lnccbrown/HSSM/assets/2658686/f607e9f8-aa4f-4827-8493-670f6b41d043">

**Comment by digicosmos86** [comment]
Hi @YuanboBQ:

Setting `link_settings` to `"log-logit"` adds link functions between the linear combination in the regressions and the response in the regressions. We added this setting so that the regressions on `a`, `z`, `t` are always within the defined domains for the likelihood function. However, doing so does change the interpretation of the regression estimates. Therefore, from how I understand the outputs, it's not surprising for some parameter values to be negative. For example, if you have a regression `y ~ ax + by`, and if you set the link function to `log`, then the regression becomes `log(y) = ax + by`.

If this is not desired behavior, we recommend that you set `link_settings` to `None`, and set `prior_settings` to `safe`. That way we apply safe priors to the regression parameters for more efficient sampling. 

I defer to @AlexanderFengler for more detailed explanation.

**Comment by YuanboBQ** [comment]
Hi, Paul
Thanks for your reply. As I described above, if I set link_settings to None, the sampling process was slow, and resulting in high values of  r_hat. I also try to updated the package to v0.2.0, and found that even using the latest version, it still result in high values of  r_hat, and when using loglik_kind = "approx_differentiable", the samling process is very slow compare to previous version.

**Comment by digicosmos86** [comment]
Will convert this into a discussion. In HSSM v0.2.1, we made some enhancements to help with the convergence issue. It will be released soon. Please try it out and let us know if that doesn't resolve this issue
